dojo-dungeon/src/dojo/rooms/rooms/riddle/Fragen_Pattern.java

```diff
@@ -131,4 +129,16 @@ private Quiz newFreeText() {
             + ".png\" dargestellt? Es reicht das Wort ohne den Zusatz Pattern!";
     return new FreeText(questionText);
   }
+
+  private void nextPattern() {
+    if (usedIndices.size() == expectedPatterns.length) {
+      usedIndices.clear();
+    }
+    int r;
+    do {
```

Comment:
das kann potentiell in eine ziemlich lang laufende schleife führen.

Reply:
> das kann potentiell in eine ziemlich lang laufende schleife führen.

Das ist richtig, wird aber erst bei sehr hohen `n` praktisch relevant.

Hätten wir 10.000 Fragen und 10.000 Schleifendurchläufe, würde mit einer Wahrscheinlichkeit von 63 % auch die letzte noch offene Frage gewählt werden.

Man kann sagen, dass diese Methode nicht mehr als `O(n)` Kosten hat, wenn `n` die Anzahl der Fragen ist.

Dennoch ändern?

Reply:
Siehe vielleicht auch: `https://www.wolframalpha.com/input?i=solve+1-%28%281000-1%29%2F1000%29%5En%3D0.5+for+n` :)